dev/com.ibm.ws.com.unboundid/src/com/ibm/ws/com/unboundid/InMemoryLDAPServer.java

```diff
@@ -0,0 +1,167 @@
+
+    private InMemoryDirectoryServerConfig config = null;
+    private InMemoryDirectoryServer ds = null;
+
+    /** The name for this server instance. */
+    private final String name = null;
+
+    /**
+     * Creates a new instance of the in memory LDAP server. It initializes the directory
+     * service.
+     *
+     * @param instance The name of this ADS instance.
+     * @throws Exception
+     *             If something went wrong
+     */
+    public InMemoryLDAPServer(String... bases) throws Exception {
+
+        config = new InMemoryDirectoryServerConfig(bases);
+        config.addAdditionalBindCredentials(getBindDN(), getBindPassword());
+        config.setListenerConfigs(
+                                  InMemoryListenerConfig.createLDAPConfig("LDAP", // Listener name
+                                                                          null, // Listen address. (null = listen on all interfaces)
+                                                                          0, // Listen port (0 = automatically choose an available port)
+                                                                          null) // StartTLS factory
+        ); // Client factory
+
+        /*
+         * Merge the default schema with our WIM schema. The WIM schema adds wimInetOrgPerson objectclass which
+         * contains all the WIM properties. Without this, the user would have to map LDAP attributes to WIM
+         * properties.
+         */
+
+        InputStream in = getClass().getResourceAsStream("/resources/wimschema.ldif");
+        Schema wimschema = Schema.getSchema(in);
+        Schema schema = Schema.mergeSchemas(Schema.getDefaultStandardSchema(), wimschema);
+        config.setSchema(schema);
+        ds = new InMemoryDirectoryServer(config);
+        ds.startListening();
+    }
+
+    /**
+     * Get the default administrative bind distinguished name.
+     *
+     * @return The default administrative bind distinguished name.
+     * @see #getBindPassword()
+     */
+    public static String getBindDN() {
+        return "uid=admin,ou=system";
+    }
+
+    /**
+     * Get the default administrative bind password.
+     *
+     * @return The default administrative bind password.
+     * @see #getBindDN()
+     */
+    public static String getBindPassword() {
+        return "secret";
+    }
+
+    /**
+     * Get the name of this LDAP server instance.
+     *
+     * @return The name.
+     */
+    public String getInstanceName() {
+        return name;
+    }
+
+    /**
+     * Get the {@link InMemoryDirectoryServer} instance.
+     *
+     * @return The {@link InMemoryDirectoryServer} instance.
+     */
+    public InMemoryDirectoryServer getLdapServer() {
+        return ds;
+    }
+
+    public void add(Entry entry) throws LDAPException {
+        ds.add(entry);
+    }
+
+    /**
+     * Stop the InMemoryDirectoryServer.
+     */
+    public void stopServer() {
+        ds.shutDown(true);
+    }
+
+    /**
```

Comment:
Shut down the InMemoryDirectoryServer.
```suggestion
    /**
     * Shut down the InMemoryDirectoryServer.
     *
```